tst/EnergyPlus/unit/ReportCoilSelection.unit.cc

```diff
@@ -0,0 +1,96 @@
+//     software as "EnergyPlus version X" software, where "X" is the version number Licensee
+//     obtained under this License and may not use a different name for the software. Except as
+//     specifically required in this Section (4), Licensee shall not use in a company name, a
+//     product name, in advertising, publicity, or other promotional activities any name, trade
+//     name, trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or confusingly
+//     similar designation, without the U.S. Department of Energy's prior written consent.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+// EnergyPlus::Standalone ERV Unit Tests
+
+// Google Test Headers
+#include <gtest/gtest.h>
+
+#include "Fixtures/EnergyPlusFixture.hh"
+#include "Fixtures/SQLiteFixture.hh"
+
+// EnergyPlus Headers
+#include <EnergyPlus/DataEnvironment.hh>
+#include <EnergyPlus/DataPrecisionGlobals.hh>
+#include <EnergyPlus/DataSizing.hh>
+#include <EnergyPlus/UtilityRoutines.hh>
+#include <EnergyPlus/DataAirSystems.hh>
+#include <EnergyPlus/DataHVACGlobals.hh>
+#include <EnergyPlus/DataZoneEquipment.hh>
+#include <EnergyPlus/Fans.hh>
+#include <EnergyPlus/HVACFan.hh>
+#include <EnergyPlus/Psychrometrics.hh>
+#include <EnergyPlus/ReportCoilSelection.hh>
+#include <EnergyPlus/ReportSizingManager.hh>
+
+using namespace EnergyPlus;
+using namespace ObjexxFCL;
+using namespace EnergyPlus::DataAirSystems;
+using namespace EnergyPlus::DataGlobals;
+using namespace EnergyPlus::DataEnvironment;
+using namespace EnergyPlus::DataHVACGlobals;
+using namespace EnergyPlus::DataSizing;
+using namespace EnergyPlus::DataZoneEquipment;
+using namespace EnergyPlus::Fans;
+using namespace EnergyPlus::Psychrometrics;
+using namespace EnergyPlus::ReportSizingManager;
+
+TEST_F( EnergyPlusFixture, ReportCoilSelection_setCoilAirFlow )
+{
+	std::string coilName ( "Coil 1" ); // user-defined name of the coil
+	std::string coilType ( "Coil:Cooling:Water" ); // idf input object class name of coil
+	Real64 airVdot ( 0.052 ); // air flow rate in m3/s
+	bool isAutoSized ( false ); // true if air flow was autosized
+
+	coilSelectionReportObj->setCoilAirFlow( coilName, coilType, airVdot, isAutoSized );
```

Comment:
So concise.  Are you adding more unit tests or is this the extent of your testing?

Reply:
From the [coverage results](http://energyplus.s3-website-us-east-1.amazonaws.com/coverage/2018-02-27-EnergyPlus-4b97a62bb0-x86_64-gcc-4.8UTCD/EnergyPlus/ReportCoilSelection.cc.gcov.html) it looks like the code that isn't covered by this test is mostly the erroneous code paths and the steam loop calculations.  However there is a section in the `setCoilWaterFlowNodeNums` function where `c->waterLoopNum > 0 && c->pltSizNum > 0` and that block doesn't get hit.  What do you think about adding some coverage there?

Reply:
This is only half-baked. Will add more.  The functions themselves are all from @EnergyArchmage so I can't take credit for how they're written. I just made copies of them to make two flavors for some.